src/lib/db/query.js

```diff
@@ -134,5 +134,20 @@ module.exports = {
           `Problem retrieving hour counts between: ${from} and ${to || 'now'} for group "${group}"`, e);
         throw new Error('Could not fetch data');
       });
-  }
+  },
+
+  cumulativeUsage: () => db.manyOrNone(
+      sqlBuilder({
+        'SELECT': 'DATE_TRUNC(\'month\', date_time) AS month,',
+             ' ': 'SUM(COUNT(*)) OVER (ORDER BY DATE_TRUNC(\'month\', date_time))::INTEGER AS count',
+        'FROM': 'lev_audit',
+        'WHERE': 'date_time < DATE_TRUNC(\'month\', NOW())',
+        'GROUP BY': '1',
```

Comment:
I think this line will make the query very slow as it will need to transform the entire `date_time` index before it can work.

Did we look at ways of doing this using 'start_of_month < date_time < end_of_month'?

I'd think something would come us if you search for 'grouping by month' or grouping by time interval.

Reply:
Depending on how slow it is we could merge for now and optimise later.

@RobinKnipe: Do you have a feel for how long it's taking?

Reply:
So, `DATE_TRUNC` still gives full timestamps, but without the unwanted (least significant) information, e.g.
```SQL
select DATE_TRUNC('minute', now()) as minute, DATE_TRUNC('day', now()) as day, DATE_TRUNC('year', now()) as year;
         minute         |          day           |          year          
------------------------+------------------------+------------------------
 2020-06-01 14:11:00+00 | 2020-06-01 00:00:00+00 | 2020-01-01 00:00:00+00
```
...which means the index still gets used.

Reply:
It will be used but it will need to run DATE_TRUNC on the whole index first I think.

Reply:
Quite possibly, but I'm at a loss to find a more performant solution, so until one presents itself... this is it!

Reply:
OK, l think we've managed to shave about 1 second off, but any further gains l think will require some form of caching.